src/main/java/org/ice4j/util/ExecutorFactory.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * ice4j, the OpenSource Java Solution for NAT and Firewall Traversal.
+ *
+ * Copyright @ 2018 Jitsi.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ice4j.util;
+
+import java.util.concurrent.*;
+
+/**
+ * Helper class which contains functions to create pre-configured executors
+ */
+public class ExecutorFactory
+{
+    /**
+     * Create {@link ScheduledExecutorService} with single executor thread
+     * @param threadNamePrefix - name prefix for threads created by pool
+     * @param threadKeepAliveTime - keep alive time before idle thread is freed
+     * @param timeUnit - time unit of <tt>threadKeepAliveTime</tt>
+     * @return pre-configured {@link ScheduledExecutorService}
+     */
+    public static ScheduledExecutorService createSingleThreadScheduledExecutor(
+        String threadNamePrefix,
+        int threadKeepAliveTime,
+        TimeUnit timeUnit)
+    {
+        return createdScheduledExecutor(
+            1,
+            threadNamePrefix,
+            threadKeepAliveTime,
+            timeUnit);
+    }
+
+    /**
+     * Create {@link ScheduledExecutorService} with number of threads up to
+     * number of CPU cores on machine
+     * @param threadNamePrefix - name prefix for threads created by pool
+     * @param threadKeepAliveTime - keep alive time before idle thread is freed
+     * @param timeUnit - time unit of <tt>threadKeepAliveTime</tt>
+     * @return pre-configured {@link ScheduledExecutorService}
+     */
+    public static ScheduledExecutorService createdCPUBoundScheduledExecutor(
```

Comment:
@bbaldino thanks for noticing typo, will fix.
Not sure if I got your question, please correct me if I’m answering wrong question. The multiple places might create multiple schedulers bound to CPU count because they might have completely different tasks which need not share same threads, but of course they share same CPU cores. They also might have different prefixes and daemon settings, and maybe priorities (when I done it).

Reply:
Sorry, I guess what I meant was: the idea behind having an executor create only as many threads as there are CPUs is that more would technically be a waste.  But, if we're creating multiple instances of that executors, we're going to end up with more threads than cores/CPUs anyway.  The CPU-bound executor just seems like it would only make sense if it were done in such a way that a single instance of it was shared by all CPU-bound code.

Reply:
@bbaldino seems like I've got what you mean now. I agree that there is ideally should not be more threads than CPU cores for CPU-bound work. Currently this method is invoked in `Agent` and `StunClientTransaction` then executor is stored in static variable, so for now only two instances actually created. If we want only one, then the `ExecutorService` is either need to be injected into constructor or read from common place, something like `ForkJoinPool.commonPool()`, but for `ice4j` it would be something like `ExecutorFactory/Utils.commonScheduleExecutorService()`. I'm not exactly sure how much non CPU bound work is actually scheduled into these `ExecutorService` right now, maybe it even make sense to have some multiple (but **fixed**) number of available CPUs if some of scheduled task actually  blocks executor thread.
Do you want me introducing shared CPU bound pool in `ice4j`s `ExecutorsFacotry` (looks like `ExecutorUtils` name already busy in libjitsi)?

BTW, fixed typo.

Reply:
No I think it's ok for now, just an observation.  I think this looks good, will merge once tests finish.